specs/vep-2272-complete-data-job-configuration-persistence/README.md

```diff
@@ -0,0 +1,161 @@
+# VEP-2272: Complete Data Job Configuration Persistence
+
+* **Author(s):** Miroslav Ivanov (miroslavi@vmware.com)
+* **Status:** draft
+
+- [Summary](#summary)
+- [Glossary](#glossary)
+- [Motivation](#motivation)
+- [Requirements and goals](#requirements-and-goals)
+- [High-level design](#high-level-design)
+- [API Design](#api-design)
+- [Detailed design](#detailed-design)
+- [Implementation stories](#implementation-stories)
+- [Alternatives](#alternatives)
+
+## Summary
+
+Currently, the deployment configuration of data jobs is partially dependent on the Kubernetes cluster, leading to
+unnecessary overhead when loading data job deployments. Additionally, the loss of Kubernetes namespace results in the
+potential loss of data jobs, requiring a manual and complex restoration process. By switching the source of truth from
+Kubernetes to a database, we can ensure a more reliable and easy restoration process while optimizing performance for
+Control Service APIs.
+
+## Glossary
+
+* VDK: https://github.com/vmware/versatile-data-kit/wiki/dictionary#vdk
+* Control Service: https://github.com/vmware/versatile-data-kit/wiki/dictionary#control-service
+* Data Job: https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job
+* Data Job Deployment: https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job-deployment
+* Kubernetes: https://kubernetes.io/
+
+## Motivation
+
+Ensuring efficient and reliable management of data job deployments is crucial for the smooth functioning of our system.
+Currently, when a data job is deployed, its deployment configuration is stored in both Kubernetes and the database.
+However, there exists a disparity where certain essential properties are exclusively stored in Kubernetes, such as the
+job's enabled status, Python version, and deployment information.
```

Comment:
I believe we currently store the enabled status, https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java#L211

Reply:
Correct

Reply:
Updated